### PR TITLE
Add notes about `ruff`.

### LINKED ISSUE #14737: DOC: Clarify that ruff standards in core do not automatically apply to coordinated and affiliated packages
Make it clear that the `ruff` checks happening in this repo (core) do not automatically apply to coordinated/affiliated packages (opt-in only).

@weaverba137 , do you have a suggestion on where is the best place in the docs for this? Thanks!

cc @nstarman and @WilliamJamieson

### PR DESCRIPTION
**Pull request by nstarman**
Fixes #14737

### CODE REVIEW

File: docs/development/codeguide.rst [hunk 1]
```diff
@@ -166,13 +166,16 @@ Coding Style/Conventions
...
-  guide. In addition, these checks also enforce
-  `isort <https://pycqa.github.io/isort/>`_ to sort the module imports.
-
-  * We provide a ``pre-commit`` hook which automatically enforces and fixes
-    (whenever possible) the coding style, see :ref:`pre-commit` for details on
-    how to setup and use this.
+* Our testing infrastructure currently enforces a subset of the PEP8 style guide. In
+  addition, these checks also enforce `isort <https://pycqa.github.io/isort/>`_ to sort
+  the module imports and a large set of style-checks supported by `ruff
+  https://beta.ruff.rs/docs/`_.
```
Comment (pllim) [inline docs/development/codeguide.rst:172]: Formatting typo?
```suggestion
  the module imports and a large set of style-checks supported by
  `ruff <https://beta.ruff.rs/docs/>`_.
```

File: docs/development/codeguide.rst [hunk 2]
```diff
@@ -166,13 +166,16 @@ Coding Style/Conventions
...
+
+  * We provide a ``pre-commit`` hook which automatically enforces and fixes (whenever
+    possible) the coding style, see :ref:`pre-commit` for details on how to setup and
```
Comment (pllim) [inline docs/development/codeguide.rst:175]:
```suggestion
    possible) the coding style, see :ref:`pre-commit` for details on how to set up and
```

**Review by mhvk — APPROVED** [review]
Yes, good to update!

**Comment by pllim** [comment]
Alas, you have to fix these 2 warnings and then squash out the commits. Don't worry about having me as co-author; doesn't matter to me in this case. Thanks!

```
docs/development/codeguide.rst:174: WARNING: py:obj reference target not found: PEP 8
docs/development/codeguide.rst:174: WARNING: py:obj reference target not found: ruff
```

**Comment by pllim** [comment]
Almost... `<rst_epilog>:174: WARNING: Duplicate explicit target name: "ruff".`

**Comment by nstarman** [comment]
> Almost... `<rst_epilog>:174: WARNING: Duplicate explicit target name: "ruff".`

That's a confusing one, because it's not complaining about PEP8, which I implemented identically to the ruff substitution.

**Comment by nstarman** [comment]
Ok. Solution worked. Thanks @mhvk & @pllim .

**Comment by pllim** [comment]
Thank you!

**Pull request merged**